codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -331,22 +331,26 @@ private Http2Stream requireStream(int streamId) {
      */
     private final class FlowControlledData extends FlowControlledBase {
         private final CoalescingBufferQueue queue;
+        private int dataSize;
 
         public FlowControlledData(Http2Stream stream, ByteBuf buf, int padding, boolean endOfStream,
                                    ChannelPromise promise) {
             super(stream, padding, endOfStream, promise);
-            queue = new CoalescingBufferQueue(promise.channel());
+            queue = new CoalescingBufferQueue(promise.channel(), 8);
```

Comment:
Yes. Previously the default size (16) happened to be selected and I'm not sure if this was intentional ( @louiscryan ?). This number sizes a queue for each data frame and it seems like most will only need a queue of size 1, and the merge frame's queue will grow accordingly.

I'm glad you brought it up bcz I wanted to discuss. Any objections to 8, should we revert, should we go lower, choose a default that can be overriden with a system property?